plugins/search/src/components/SearchBar/SearchBar.tsx

```diff
@@ -31,6 +31,19 @@ export const SearchBar = ({ className, debounceTime = 0 }: Props) => {
   const { term, setTerm } = useSearch();
   const [value, setValue] = useState<string>(term);
 
+  // Use useRef instead of usePrevious here since valueRef is not required
+  // to be listed as a dep of the useEffect hook so we can detect and ignore when
+  // value has changed but term has not yet been updated due to the debounce without
+  // taking into account changes to valueRef triggering the hook
+  const valueRef = useRef(value);
+  useEffect(() => {
+    // Only respond to changes to term
+    if (valueRef.current === value && value !== term) {
+      setValue(term);
+    }
+    valueRef.current = value;
+  }, [term, value]);
+
```

Comment:
We already have access to the previous value by passing a function to `setValue`, so testing locally it works without refs and the state will not be updated if it returns the previous value (else): 

```suggestion
  useEffect(() => {
    setValue(prevValue => prevValue !== term ? term : prevValue);
  }, [term]);
```

Reply:
ah thanks for the tip! definitely looks a cleaner this way